src/utils/auth.ts

```diff
@@ -0,0 +1,30 @@
+import express, {Request, NextFunction} from 'express';
+const jwt = require('jsonwebtoken');
+import { PrismaClient } from '@prisma/client';
+
+const prisma = new PrismaClient();
+
+export interface ReqWithUser extends Request {
+    user: {}
```

Comment:
Can you add the user type from Prisma maybe? This is typing user as an empty object (I'm surprised that TS is ok with that)

You shouldbe able to do:
```
import type { User } from '@prisma/client'
```

(if that doesn't work, try to add a custom type with the fields expected on user - username, email, id)